### ISSUE TITLE
Breaking 3.4.0 Typescript Changes for Extensions

### ISSUE THREAD

**Issue by wyattjoh** [kamilkisiela/graphql-config#716]
In the 3.4.0 release, you bumped the version of `@graphql-tools/utils` to `8.0.0` (it's not in the GitHub repository for some reason?). This caused a breaking change with existing extensions that could previously have been used with the `loadConfig` function:

```ts
import { CodegenExtension } from "@graphql-codegen/cli";
import { loadConfig } from "graphql-config";

const config = await loadConfig({
  extensions: [CodegenExtension],
});
```

This is an issue because the `GraphQLExtensionDeclaration` type exported does expose an indirect reference to the type from:

```ts
import { Loader } from '@graphql-tools/utils';
```

The issue here being that the `Loader` type signature changed in version `7.10.0` to `8.0.0` (as seen here https://github.com/ardatan/graphql-tools/compare/@graphql-tools/utils@7.10.0...@graphql-tools/utils@8.0.0#diff-c53c3ed3e49523c6f542babc8c8e2bc105d0a27f7d6aca43e596272f766d6bf0). I'd suggest rolling back the dependancy upgrade for `@graphql-tools/utils` to `7.10.0`, cutting a new `3.4.1` release, and then doing the upgrade to `8.0.0` via a major version bump to `4.0.0`

**To Reproduce**

You can reproduce this with the following versions and code:

```json
{
  "devDependencies": {
    "@graphql-codegen/cli": "^1.21.8",
    "graphql-config": "^3.4.0"
  }
}
```

```ts
import { CodegenExtension } from "@graphql-codegen/cli";
import { loadConfig } from "graphql-config";

const config = await loadConfig({
  extensions: [CodegenExtension],
});
```

Where via typescript, you can see the type error:

```
Type 'import("/<redacted>/node_modules/@graphql-codegen/cli/node_modules/graphql-config/extension").GraphQLExtensionDeclaration' is not assignable to type 'import("/<redacted>/node_modules/graphql-config/extension").GraphQLExtensionDeclaration'.
  Types of parameters 'api' and 'api' are incompatible.
    Type 'import("/<redacted>/node_modules/graphql-config/extension").ExtensionAPI' is not assignable to type 'import("/<redacted>/node_modules/@graphql-codegen/cli/node_modules/graphql-config/extension").ExtensionAPI'.
      The types of 'loaders.schema.register' are incompatible between these types.
        Type '(loader: Loader<BaseLoaderOptions>) => void' is not assignable to type '(loader: Loader<string, SingleFileOptions>) => void'.
          Types of parameters 'loader' and 'loader' are incompatible.
            Type 'Loader<string, SingleFileOptions>' is not assignable to type 'Loader<BaseLoaderOptions>'.
              The types returned by 'load(...)' are incompatible between these types.
                Type 'Promise<Source>' is not assignable to type 'Promise<Source[] | null>'.
                  Type 'Source' is missing the following properties from type 'Source[]': length, pop, push, concat, and 28 more.
```

**Environment:**

- OS: macOS 11.5.1
- GraphQL Config Version: 3.4.0
- NodeJS: 14.15.4

**Comment by dotansimha** [comment]
Thanks @wyattjoh for reporting this. We did miss this breaking change. 
@kamilkisiela will release `3.4.1` with an API adjustment, and we released `4.0.0` with a note for this breaking change.

**Comment by kamilkisiela** [comment]
6c7e680469389f8db43c7995cb346c21ec70ff2f
